python/hail/expr/expression.py

```diff
@@ -1240,19 +2346,63 @@ class Int64Expression(NumericExpression):
 
 
 class StringExpression(AtomicExpression):
-    """Expression of type :class:`hail.expr.types.TString`."""
+    """Expression of type :class:`hail.expr.types.TString`.
+
+    >>> s = functions.capture('The quick brown fox')
+    """
     def __getitem__(self, item):
+        """Slice or index into the string.
+
+        Examples
+        --------
+        .. doctest::
+
+            >>> eval_expr(s[:15])
+            'The quick brown'
+
+            >>> eval_expr(s[0])
+            'T'
+
+        Parameters
+        ----------
+        item : slice or :class:`Int32Expression`
+            Slice or character index.
+
+        Returns
+        -------
+        :class:`StringExpression`
+            Substring or character at index `item`.
+        """
         if isinstance(item, slice):
             return self._slice(TString(), item.start, item.stop, item.step)
-        elif isinstance(item, int):
+        elif isinstance(item, int) or isinstance(item, Int32Expression):
```

Comment:
bugfix here too

Reply:
what is self._index? can you just pass it a expression like that? (also can a slice have an expression in it?)

Reply:
Here are the definitions of those two guys:
```python
    def _index(self, ret_typ, key):
        key = to_expr(key)
        indices, aggregations, joins = unify_all(self, key)
        return construct_expr(Index(self._ast, key._ast),
                              ret_typ, indices, aggregations, joins)

    def _slice(self, ret_typ, start=None, stop=None, step=None):
        if start is not None:
            start = to_expr(start)
            start_ast = start._ast
        else:
            start_ast = None
        if stop is not None:
            stop = to_expr(stop)
            stop_ast = stop._ast
        else:
            stop_ast = None
        if step is not None:
            raise NotImplementedError('Variable slice step size is not currently supported')

        non_null = [x for x in [start, stop] if x is not None]
        indices, aggregations, joins = unify_all(self, *non_null)
        return construct_expr(Index(self._ast, Slice(start_ast, stop_ast)),
                              ret_typ, indices, aggregations, joins)
```

Reply:
You can indeed pass an expression into an index as well as a slice!